library/src/G4CMPProcessUtils.cc

```diff
@@ -132,8 +139,9 @@ void G4CMPProcessUtils::LoadDataForTrack(const G4Track* track) {
 
     const G4ParticleDefinition* pd = track->GetParticleDefinition();
     G4Track* tmp_track = const_cast<G4Track*>(track);
-    tmp_track->SetMomentumDirection(
-      theLattice->MapKtoVDir(G4PhononPolarization::Get(pd), tempvec));
+    const G4ThreeVector localVDir = theLattice->MapKtoVDir(G4PhononPolarization::Get(pd),
```

Comment:
I don't want to use a per-call temporary here.  We have so many of these calls in the job that it could make a time difference.  

Since you're already using tempvec for the k vector, I think maybe we should add a`tempvec2` data member.  It's ugly, but will improve performance (so that should be part of the comment associated with it).